integration/tests/session_key_spending.test.ts

```diff
@@ -0,0 +1,372 @@
+import { BlockchainUtils } from '@/blockchainUtils';
+import { DatabaseUtils } from '@/databaseUtils';
+import { Identity } from '@/identity';
+import { TestNitroliteClient } from '@/nitroliteClient';
+import { TestWebSocket } from '@/ws';
+import { RPCAppStateIntent, RPCProtocolVersion } from '@erc7824/nitrolite';
+import { Hex } from 'viem';
+import { setupTestIdentitiesAndConnections } from '@/testSetup';
+import {
+    createTestChannels,
+    authenticateAppWithAllowances,
+    createTestAppSession,
+    toRaw,
+    getLedgerBalances,
+} from '@/testHelpers';
+import { submitAppStateUpdate_v04 } from '@/testAppSessionHelpers';
+
+describe('Session Key Spending Caps', () => {
+    const onChainDepositAmount = BigInt(1000);
+    const spendingCapAmount = BigInt(500); // Session key limited to 500 USDC
+    const initialDepositAmount = BigInt(100);
+
+    let aliceWS: TestWebSocket;
+    let alice: Identity;
+    let aliceClient: TestNitroliteClient;
+
+    let aliceAppWS: TestWebSocket;
+    let aliceAppIdentity: Identity;
+
+    let bobWS: TestWebSocket;
+    let bob: Identity;
+    let bobAppIdentity: Identity;
+    let bobClient: TestNitroliteClient;
+
+    let blockUtils: BlockchainUtils;
+    let databaseUtils: DatabaseUtils;
+
+    let appSessionId: string;
+
+    let currentVersion = 1;
+
+    const SESSION_DATA = { gameType: 'chess', gameState: 'waiting' };
+
+    beforeAll(async () => {
+        blockUtils = new BlockchainUtils();
+        databaseUtils = new DatabaseUtils();
+
+        const { Identity: IdentityClass } = await import('@/identity');
+        const { TestWebSocket: TestWebSocketClass } = await import('@/ws');
+        const { TestNitroliteClient: TestNitroliteClientClass } = await import('@/nitroliteClient');
+        const { createAuthSessionWithClearnode } = await import('@/auth');
+        const { CONFIG } = await import('@/setup');
+
+        alice = new IdentityClass(CONFIG.IDENTITIES[0].WALLET_PK, CONFIG.IDENTITIES[0].SESSION_PK);
+        aliceWS = new TestWebSocketClass(CONFIG.CLEARNODE_URL, CONFIG.DEBUG_MODE);
+        aliceClient = new TestNitroliteClientClass(alice);
+        await aliceWS.connect();
+        await createAuthSessionWithClearnode(aliceWS, alice);
+
+        aliceAppIdentity = new IdentityClass(CONFIG.IDENTITIES[0].WALLET_PK, CONFIG.IDENTITIES[0].APP_SESSION_PK);
+        aliceAppWS = new TestWebSocketClass(CONFIG.CLEARNODE_URL, CONFIG.DEBUG_MODE);
+        await aliceAppWS.connect();
+
+        bob = new IdentityClass(CONFIG.IDENTITIES[1].WALLET_PK, CONFIG.IDENTITIES[1].SESSION_PK);
+        bobAppIdentity = new IdentityClass(CONFIG.IDENTITIES[1].WALLET_PK, CONFIG.IDENTITIES[1].APP_SESSION_PK);
+        bobWS = new TestWebSocketClass(CONFIG.CLEARNODE_URL, CONFIG.DEBUG_MODE);
+        bobClient = new TestNitroliteClientClass(bob);
+        await bobWS.connect();
```

Comment:
You can use `setupTestIdentitiesAndConnections` for all these initializations in the same way as it is done in `lifecycle_nitrorpc_v04.test.ts`.